### ISSUE TITLE
Swupd update are leaving files behind after an update

### ISSUE THREAD

**Issue by otaviobp** [clearlinux/mixer-tools#750]
**Describe the bug**

After an update some files that should be removed are not being removed.

**To Reproduce**
```
# sudo swupd os-install --path /opt/a -B gnome-base-libs -V 32480
# sudo swupd repair --extra-files-only --path /opt/a --no-scripts # no extra file found
# sudo swupd update --path /opt/a --no-scripts
# sudo swupd repair --extra-files-only --path /opt/a --no-scripts # Multiple extra files found
```

**Expected behavior**
 - No files should be left behind
 - We should have tests implemented to make sure this was really fixed

**Environment (please complete the following information):**
 - Clear Linux OS Version: 32510
 - Platform: Any

Problem reported by @rossburton on clearlinux/swupd-client#1346. More information available at that issue

**Comment by gtkramer** [comment]
Another way to reproduce this:
* Download https://cdn.download.clearlinux.org/releases/30880/clear/clear-30880-kvm.img.xz
* Extract image and boot with kvm
* swupd bundle-add kvm-host python-extras koji
* swupd clean --all
* swupd repair --picky (comes out basically clean)
* swupd clean --all
* while swupd check-update; do swupd update; done
* swupd clean --all
* swupd repair --picky

The last command removes a bunch of files that did not get removed as a part of the update process.  It should come out with pretty much the same extra files as the first diagnose.  Always working with a clean cache is a bit defensive, but keeps the cache aspect out of this.

**Comment by jwakre** [comment]
These leftover files are caused by a process error where the format number was accidentally incremented and rolled back. In v32480 the format number was accidentally incremented to 30 and in v32500 the format number was rolled back to 29. When the format number was incremented, the deleted file entries were dropped from the manifest. Future versions will not re-add the deleted file entries.

Steps to reproduce:

```
mixer init --format 30
sudo mixer build all --increment # initial version
echo filesystem > local-bundles/os-core
sudo mixer build all --increment # version with deleted files
sudo mixer build all --increment --format 31 # increment format number, drop deleted entries
sudo mixer build all --increment --format 30 # roll-back format number, deleted file entries are still dropped
```

Since deleted files are expected to be dropped after a format bump, this is expected Mixer behavior.

@gtkramer That was a valid issue, but it was fixed by #684
